packages/adp-tooling/src/base/project-utils.ts

```diff
@@ -0,0 +1,206 @@
+import { join } from 'path';
+import type { Editor } from 'mem-fs-editor';
+import { AdpWriterConfig, CfAdpConfig, CfModuleData, ManifestAppdescr, ProjectType } from '../types';
+import {
+    enhanceUI5DeployYaml,
+    enhanceUI5Yaml,
+    hasDeployConfig,
+    enhanceUI5YamlWithCustomConfig,
+    enhanceUI5YamlWithCustomTask
+} from '../writer/options';
+import { UI5Config } from '@sap-ux/ui5-config';
+
+/**
+ * Writes a given project template files within a specified folder in the project directory.
+ *
+ * @param {string} templatePath - The root path of the project template.
+ * @param {string} projectPath - The root path of the project.
+ * @param {CfModuleData | AdpWriterConfig} data - The data to be populated in the template file.
+ * @param {Editor} fs - The `mem-fs-editor` instance used for file operations.
+ * @param {string[]} exclude - Files to be ignored when writing.
+ * @returns {void}
+ */
+export function writeTemplateToFolder(
+    templatePath: string,
+    projectPath: string,
+    data: CfModuleData | AdpWriterConfig,
+    fs: Editor,
+    exclude: string[] | [] = []
```

Comment:
Maybe as a suggestion for more clarity, this should be `ignoredFiles: string[] ...` or `excludedFiles`